clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2737,6 +2737,29 @@ void SYCLIntegrationHeader::emit(raw_ostream &O) {
 
   O << "\n";
 
+  // Get c++ version to support version mismatch diagnostics
+  int CppVersion = -1;
+  LangOptions LangOpts = S.getASTContext().getLangOpts();
+  assert(LangOpts.CPlusPlus);
+  if (LangOpts.CPlusPlus20)
+    CppVersion = 202002L;
+  else if (LangOpts.CPlusPlus17)
+    CppVersion = 201703L;
+  else if (LangOpts.CPlusPlus14)
+    CppVersion = 201402L;
+  else if (LangOpts.CPlusPlus11)
+    CppVersion = 201103L;
+  else
+    CppVersion = 199711L;
```

Comment:
I think that @erichkeane mentioned that it shouldn't be a **member** function. I think we can keep it as some function available in SemaSYCL .